src/features/myAccounts/hooks/useAllOwnedSafes.ts

```diff
@@ -17,7 +17,7 @@ const useAllOwnedSafes = (address: string): AsyncResult<AllOwnedSafes> => {
     if (data != undefined) {
       setCache(data)
     }
-  }, [address, cache, data, setCache])
+  }, [data, setCache])
```

Comment:
As per @compojoom comment, should we restore `address` here?
Only `cache` was causing a recursion.

Reply:
I don't think it's necessary as keeping `address` as a dep. results in two `setCache` calls each time it changes:

1. The first happens immediately when `address` changes.
2. The second happens when `useGetAllOwnedSafesQuery` fetches and updates `data`.

Reply:
`data` remains populated when the wallet is disconnected though. [Passing `skipToken` to `useGetAllOwnedSafesQuery`](https://github.com/safe-global/safe-wallet-web/blob/c33e74a9c44b1d15a409091ea62a561f585170b2/src/features/myAccounts/hooks/useAllOwnedSafes.ts#L14) prevents the query from fetching new data or clearing the existing cached data.

Reply:
OK, then let's treat it as a separate bug. I'll merge the PR now, thanks a lot!